objects/c4e8cd/script.lua

```diff
@@ -1 +1,977 @@
 difficulty=-1
+postSetup = true
+postSetupComplete = false
+hasBroadcast = true
+
+local destinyPlayers = {}
+local readyTokens = {}
+local round = 1 -- First or second round of drafts
+local phase = 0 -- Phase within a round
+local timer = nil
+
+function onSave()
+    local data_table = {
+        destinyPlayers = destinyPlayers,
+        readyTokens = readyTokens,
+        round = round,
+        phase = phase,
+    }
+    return JSON.encode(data_table)
+end
+
+function onLoad(saved_data)
+    if saved_data ~= "" then
+        local loaded_data = JSON.decode(saved_data)
+        destinyPlayers = loaded_data.destinyPlayers
+        readyTokens = loaded_data.readyTokens
+        round = loaded_data.round
+        phase = loaded_data.phase
+    end
+
+    Color.Add("SoftBlue", Color.new(0.53,0.92,1))
+
+    if Global.getVar("gameStarted") and Global.getVar("scenarioCard") ~= nil and Global.getVar("scenarioCard").guid == self.guid then
+        PostSetup()
+    end
+end
+
+function Broadcast()
+    return "Use the button on the scenario panel to progress drafting"
+end
+
+function PostSetup(_)
+    if round <= 2 then
+        -- Lock out card gains for the duration
+        Wait.condition(
```

Comment:
i feel like there's something i'm missing here but this just sets the variable true one time? how does that tie into the "duration" part?

Reply:
It gets set false again in `cleanUp()`, once the whole Destiny Unfolds drafting business is done. Basically, the point of this code is to prevent any of the "Gain a Major/Minor" buttons from working for the entire duration of the time Destiny Unfolds is doing its scripted thing, to avoid them clearing the power zones and messing everything up. For this purpose, I'm using the Global `scriptWorkingCardC` variable, which is otherwise used by the global code for double-click protection on the "Gain a Major/Minor".

Reply:
ahh i see how this works now, thanks